test-basic/documents-data-movement-readAll.js

```diff
@@ -216,19 +235,179 @@ describe('data movement readAll', function() {
                 if(err){
                     done(err);
                 }
-                arr.forEach(item=> result.add(item.uri)
-                );
+                arr.forEach(item=> result.add(item.uri));
                 checkResult(done);
             });
     });
+
+    it('should readAll documents with categories option', function(done){
+        uriStream = new Stream.PassThrough({objectMode: true});
+        let readable = new Stream.Readable({objectMode: true});
+
+        for (let i = 0; i < 400; i++) {
+            const temp = {
+                uri: '/test/dataMovement/requests/readAll/categories/' + i + '.json',
+                contentType: 'application/json',
+                content: {['key ' + i]: 'value ' + i}
+            };
+            readable.push(temp);
+            uriStream.push(temp.uri);
+        }
+        readable.push(null);
+        uriStream.push(null);
+
+        readable.pipe(dbWriter.documents.writeAll({
+            defaultMetadata: {
+                metadataValues: {
+                    'metadataKey1': 'metadataValue 1',
+                    'metadataKey2': 'metadataValue 2'
+                },
+                collections: 'collection1',
+                permissions: [
+                    {'role-name': 'app-user', capabilities: ['read']},
+                    {'role-name': 'app-builder', capabilities: ['read', 'update']}
+                ],
+                quality: 1
+            }
+        }));
```

Comment:
If I recall correctly, Mocha has a before task that could do the setup separately from the it test, which would be better than using a timeout to execute the test after the setup.  

Or, the test could move into an on completion callback for the setup similar to some of the other tests to execute in a result callback.

(The concern is that the timeout could be fragile.)